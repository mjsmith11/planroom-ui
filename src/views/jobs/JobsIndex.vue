<template>
<div class="container">
    <div class="col-xs-12 col-sm-10 col-md-8 offset-md-2 offset-sm-1">
        <h1>Jobs</h1>
        <div class="list-group">
            <job-item v-for="job in jobs" :job="job" :key="job.id"></job-item>
        </div>
    </div>
</div>
</template>
<script>
import axios from 'axios'
import JobItem from '../../components/jobs/ListItem.vue'
export default {
  data () {
    return {
      jobs: []
    }
  },
  components: {
    JobItem
  },
  created () {
    axios.get('/jobs')
      .then(res => {
        console.log(res)
        this.jobs = res.data
      })
      .catch(err => {
        console.log(err)
      })
  }
}
</script>
<style scoped>

</style>
