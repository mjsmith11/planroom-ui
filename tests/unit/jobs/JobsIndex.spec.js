import { mount, RouterLinkStub } from '@vue/test-utils'
import JobsIndex from '@/views/jobs/JobsIndex'
import mockAxios from 'jest-mock-axios'

describe('Job Index', () => {
  let cmp

  beforeEach(() => {
    cmp = mount(JobsIndex, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
  })

  afterEach(() => {
    // cleaning up the mess left behind the previous test
    mockAxios.reset()
  })

  it('has expected html structure', () => {
    const response = {
      data: [
        {
          'bidDate': '2017-03-23',
          'bidEmail': 'abc@xyz.com',
          'bonding': true,
          'name': 'abc',
          'prebidAddress': '123 Main St.',
          'prebidDateTime': '2018-07-01T18:30',
          'subcontractorBidsDue': '2017-06-01T08:30',
          'taxible': true,
          'id': 25
        },
        {
          'bidDate': '2017-12-23',
          'bidEmail': 'abcdef@xyz.com',
          'bonding': false,
          'name': 'def',
          'prebidAddress': '234 Main St.',
          'prebidDateTime': '2018-07-01T18:30',
          'subcontractorBidsDue': '2017-06-01T08:30',
          'taxible': true,
          'id': 20
        }
      ]
    }
    mockAxios.mockResponse(response)
    expect(cmp.html()).toMatchSnapshot()
  })

  it('processes api response', () => {
    expect(mockAxios.get).toHaveBeenCalledWith('/jobs')
    expect(mockAxios.get).toHaveBeenCalledTimes(1)

    const response = {
      data: [
        {
          'bidDate': '2017-03-23',
          'bidEmail': 'abc@xyz.com',
          'bonding': true,
          'name': 'My Test Job',
          'prebidAddress': '123 Main St.',
          'prebidDateTime': '2018-07-01T18:30',
          'subcontractorBidsDue': '2017-06-01T08:30',
          'taxible': true,
          'id': 25
        },
        {
          'bidDate': '2017-12-23',
          'bidEmail': 'abcdef@xyz.com',
          'bonding': false,
          'name': 'My Second Test Job',
          'prebidAddress': '234 Main St.',
          'prebidDateTime': '2018-07-01T18:30',
          'subcontractorBidsDue': '2017-06-01T08:30',
          'taxible': true,
          'id': 20
        }
      ]
    }
    mockAxios.mockResponse(response)
    expect(cmp.vm.jobs).toBe(response.data)
  })

  // TODO add tests for error handling when it's implemented
})
