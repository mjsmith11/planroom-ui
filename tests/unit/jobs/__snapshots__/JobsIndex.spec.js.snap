// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Index has expected html structure 1`] = `
<div class="container">
  <div class="col-xs-12 col-sm-10 col-md-8 offset-md-2 offset-sm-1">
    <h1>Jobs</h1>
    <div class="list-group"><a class="list-group-item-action list-group-item list-group-item-dark">
        <div class="row">
          <div class="col-lg-9 col-xl-10">abc</div>
          <div class="col-lg-3 col-xl-2 d-none d-lg-block"><strong>03/23/2017</strong></div>
        </div>
      </a><a class="list-group-item-action list-group-item list-group-item-dark">
        <div class="row">
          <div class="col-lg-9 col-xl-10">def</div>
          <div class="col-lg-3 col-xl-2 d-none d-lg-block"><strong>12/23/2017</strong></div>
        </div>
      </a></div>
  </div>
</div>
`;
